crypto/provider_conf.c

```diff
@@ -89,7 +89,7 @@ static int provider_conf_params_internal(OSSL_PROVIDER *prov,
          */
         for (i = 0; i < sk_OPENSSL_CSTRING_num(visited); i++) {
             if (sk_OPENSSL_CSTRING_value(visited, i) == value) {
-                ERR_raise(ERR_LIB_CRYPTO, ERR_R_INTERNAL_ERROR);
+                ERR_raise(ERR_LIB_CRYPTO, CONF_R_RECRSIVE_SECTION_REFERENCE);
```

Comment:
You need to use ERR_LIB_CONF for CONF_R_.... Also there is a typo CONF_R_RECURSIVE_SECTION_REFERENCE.

Although this is not in the conf subdirectory I do not think it matters much as I think this reason code logically belongs to CONF.